src/components/Scrollytelling.astro

```diff
@@ -0,0 +1,275 @@
+---
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import maplibregl from "maplibre-gl";
+import type {
+  FillLayerSpecification,
+  CircleLayerSpecification,
+  LineLayerSpecification,
+  SymbolLayerSpecification,
+} from "maplibre-gl";
+import fs from "fs";
+import yaml from "js-yaml";
+
+// Load and parse the YAML file
+const fileContents = fs.readFileSync(
+  "src/pages/_scrollytelling_steps.yaml",
+  "utf8"
+);
+interface MapStep {
+  title: string;
+  description: string;
+  center: [number, number];
+  zoom: number;
+  pitch?: number;
+  bearing?: number;
+  media?: string;
+  position?: "left" | "right" | "centered" | "full";
+  layers?: GeoJSONFeatureLayer[];
+  persist?: boolean;
+}
+
+const mapData: MapStep[] = yaml.load(fileContents) as MapStep[];
+
+export interface Props {
+  latitude: number;
+  longitude: number;
+  zoom: number;
+  mapstyle: string;
+  container: string;
+
+  /** If `false`, the map will not respond to interaction.  This is a static map built with the full web map rendering API. */
+  interactive?: boolean;
+  containerstyle?: string;
+  pitch?: number;
+  bearing?: number;
+  layers?: GeoJSONFeatureLayer[];
+}
+
+export interface GeoJSONFeatureLayer {
+  dataType: "geojson";
```

Comment:
i'm down with the `-` separator instead of camel case for these variables but we should apply them consistently throughout each map type